sui-execution/latest/sui-adapter/src/programmable_transactions/data_store.rs

```diff
@@ -97,3 +100,29 @@ impl DataStore for SuiDataStore<'_, '_> {
         Ok(())
     }
 }
+
+// A unifying trait that allows us to load move packages that may not be objects just yet (e.g., if
+// they were published in the current transaction). Note that this needs to load `MovePackage`s and
+// not `MovePackageObject`s.
+pub trait PackageStore {
+    fn get_package(&self, id: &ObjectID) -> SuiResult<Option<Rc<MovePackage>>>;
```

Comment:
This will be used in future PRs as well. This will be quite useful for e.g., caching layers and a lot more ;)